oidc/verify.go

```diff
@@ -21,6 +21,25 @@ const (
 	issuerGoogleAccountsNoScheme = "accounts.google.com"
 )
 
+// TokenExpiredError indicates that Verify failed because the token was expired. This
+// error does NOT indicate that the token is not also invalid for other reasons. Other
+// checks might have failed if the expiration check had not failed.
+type TokenExpiredError struct {
+	// Expiry is the time when the token expired.
+	Expiry time.Time
+}
+
+func (e *TokenExpiredError) Error() string {
+	return fmt.Sprintf("oidc: token is expired (Token Expiry: %v)", e.Expiry)
+}
+
+// Is returns true if the error was due to an expired token. It does not check that
+// the expiry time is identical.
+func (e *TokenExpiredError) Is(target error) bool {
```

Comment:
Implementing is shouldn't be necessary. Can you drop this? https://go.dev/play/p/fTGu3-8H819

Reply:
Apparently `errors.Is` is doing string comparison. Since the Expiry is included in the time, I have to use a custom `Is` to avoid checking the Expiry.

Reply:
Demo: https://go.dev/play/p/7rN5_y9ODKD